src/evict/evict_page.c

```diff
@@ -675,12 +675,42 @@ __evict_review(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t evict_flags, bool
             LF_SET(WT_REC_SCRUB);
     }
 
-    /* Reconcile the page. */
-    ret = __wt_reconcile(session, ref, NULL, flags);
+    /* Acquire a snapshot if coming through eviction thread route. */
+    if (F_ISSET(session, WT_SESSION_INTERNAL) && !WT_SESSION_BTREE_SYNC(session) &&
```

Comment:
Can we simplify this check by introducing a session flag for eviction threads?

Reply:
In one the conversations @agorrod mentioned it might not be a good idea to make sessions aware that they are working in eviction context. I can add flag which uniquely identifies the sessions created for eviction threads but I can't see any use beyond this particular scenario.

Reply:
@hbokhari your response here gives another reason to avoid it. Would the flag indicate that the session is *currently* being used for eviction, or is a thread dedicated to eviction?
OTOH: This is a very complex condition, which probably makes it fragile. How did you come up with the set of conditions? Could you describe them in a comment?

Reply:
@agorrod The flag would indicate that this session was created specifically for eviction workers threads, and therefore will never be set in any other context (application threads, checkpoint).

To best of my knowledge, only eviction threads, application threads or checkpoints can enter this part of code.

Reply:
@agorrod I have used a new flag in 2257db5 to simplify this `if` condition.